src/display/network.js

```diff
@@ -420,6 +431,22 @@ class PDFNetworkStreamRangeRequestReader {
     this.onClosed = null;
   }
 
+  _onHeadersReceived() {
+    const responseOrigin = getResponseOrigin(
+      this._manager.getRequestXhr(this._requestId)?.responseURL
+    );
+
+    if (responseOrigin !== this._manager._responseOrigin) {
+      this._readCapability.reject(
```

Comment:
You don't need a separate `_readCapability` promise if you reject all pending `requests` upon setting the error. Here is a patch that you can apply on top of your current patch. It deletes some code, and tweaks `onError` so that it does not overwrite an existing error (because canceling a request is going to generate a new error and we shouldn't replace the error to avoid confusion - perhaps that should be a comment in the code?). It still passes the existing unit tests:

```diff
diff --git a/src/display/network.js b/src/display/network.js
index e4d1264a3..3beb3257b 100644
--- a/src/display/network.js
+++ b/src/display/network.js
@@ -421,7 +421,6 @@ class PDFNetworkStreamRangeRequestReader {
     };
     this._url = manager.url;
     this._requestId = manager.requestRange(begin, end, args);
-    this._readCapability = Promise.withResolvers();
     this._requests = [];
     this._queuedChunk = null;
     this._done = false;
@@ -437,14 +436,11 @@ class PDFNetworkStreamRangeRequestReader {
     );
 
     if (responseOrigin !== this._manager._responseOrigin) {
-      this._readCapability.reject(
-        new Error(
-          `Expected range response-origin "${responseOrigin}" to match "${this._manager._responseOrigin}".`
-        )
+      this._storedError = new Error(
+        `Expected range response-origin "${responseOrigin}" to match "${this._manager._responseOrigin}".`
       );
-      return;
+      this._onError(0);
     }
-    this._readCapability.resolve();
   }
 
   _close() {
@@ -468,8 +464,7 @@ class PDFNetworkStreamRangeRequestReader {
   }
 
   _onError(status) {
-    this._storedError = createResponseStatusError(status, this._url);
-    this._readCapability.reject(this._storedError);
+    this._storedError ??= createResponseStatusError(status, this._url);
     for (const requestCapability of this._requests) {
       requestCapability.reject(this._storedError);
     }
@@ -488,8 +483,6 @@ class PDFNetworkStreamRangeRequestReader {
   }
 
   async read() {
-    await this._readCapability.promise;
-
     if (this._storedError) {
       throw this._storedError;
     }
```